src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,37 @@
 package core.basesyntax.service;
 
+import core.basesyntax.cases.of.exceptions.ExceptionDuringRegistration;
+import core.basesyntax.cases.of.exceptions.NullExceptionDuringRegistration;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_YEAR = 18;
+    private static final int MIN_LENGTH = 6;
+    private static final String USER_NULL = "The user can't be Null";
+    private static final String USER_EXISTS = "User exists";
+    private static final String AGE_UNDER_18 = "Age - over " + MIN_YEAR + " years old";
+    private static final String INCORRECT_LOGIN_OR_PASSWORD = "Login and password must contain "
+            + "at least " + MIN_LENGTH + " characters";
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new NullExceptionDuringRegistration(USER_NULL);
+        }
+        if (user.equals(storageDao.get(user.getLogin()))) {
```

Comment:
I think we should not find user by null login.
`equals` is redundant. If storageDao returns some entity that indicates that the user already exists.